File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/AccountInfoData.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AccountInfoData._

case class AccountInfoData (
  group: Option[String],
address: Option[String],
city: Option[String],
country: Option[String],
disable_cc: Option[String],
fraudrecord_score: Option[String],
ima: Option[String],
name: Option[String],
payment_method: Option[String],
phone: Option[String],
pin: Option[String],
state: Option[String],
status: Option[String],
zip: Option[String],
account_id: Option[String],
account_lid: Option[String],
address2: Option[String],
affiliate_dock_description: Option[String],
affiliate_dock_title: Option[String],
affiliate_payment_method: Option[String],
affiliate_paypal: Option[String],
cc: Option[String],
cc_auto: Option[String],
cc_exp: Option[String],
cc_type: Option[String],
cc_whitelist: Option[String],
ccs: Option[AccountInfoDataCcs],
ccs_added: Option[String],
company: Option[String],
currency: Option[String],
disable_reinstall: Option[String],
disable_reset: Option[String],
email: Option[String],
email_abuse: Option[String],
email_settings: Option[AccountInfoDataEmailSettings],
extra: Option[AccountInfoDataExtra],
facebook_id: Option[String],
facebook_url: Option[String],
firstname: Option[String],
fraudrecord: Option[AccountInfoDataFraudrecord],
github_id: Option[String],
github_url: Option[String],
google_id: Option[String],
google_url: Option[String],
innertell_id: Option[String],
lastname: Option[String],
locale: Option[String],
maxmind: Option[AccountInfoMaxMindResponse],
maxmind_score: Option[String],
mb_id: Option[String],
modernbill_id: Option[String],
picture: Option[String],
referrer_coupon: Option[String],
reseller_markup: Option[String],
username: Option[String],
ssh_key: Option[String],
ssh_key_wrapped: Option[String],
api_key: Option[String],
api_key_wrapped: Option[String],
`2fa_google_key`: Option[String],
`2fa_google_enabled`: Option[Boolean],
`2fa_google`: Option[Integer],
`2fa_google_split`: Option[String],
`2fa_google_qr`: Option[String])

object AccountInfoData {
  import DateTimeCodecs._

  implicit val AccountInfoDataCodecJson: CodecJson[AccountInfoData] = CodecJson.derive[AccountInfoData]
  implicit val AccountInfoDataDecoder: EntityDecoder[AccountInfoData] = jsonOf[AccountInfoData]
  implicit val AccountInfoDataEncoder: EntityEncoder[AccountInfoData] = jsonEncoderOf[AccountInfoData]
}

File: OPENAPI-REST-API/openapi-client/clojure/src/inter_server_management_api/specs/account_info_data.clj
(ns inter-server-management-api.specs.account-info-data
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [inter-server-management-api.specs.account-info-data-ccs :refer :all]
            [inter-server-management-api.specs.account-info-data-email-settings :refer :all]
            [inter-server-management-api.specs.account-info-data-extra :refer :all]
            [inter-server-management-api.specs.account-info-data-fraudrecord :refer :all]
            [inter-server-management-api.specs.account-info-max-mind-response :refer :all]
            )
  (:import (java.io File)))


(def account-info-data-data
  {
   (ds/opt :group) string?
   (ds/opt :address) string?
   (ds/opt :city) string?
   (ds/opt :country) string?
   (ds/opt :disable_cc) string?
   (ds/opt :fraudrecord_score) string?
   (ds/opt :ima) string?
   (ds/opt :name) string?
   (ds/opt :payment_method) string?
   (ds/opt :phone) string?
   (ds/opt :pin) string?
   (ds/opt :state) string?
   (ds/opt :status) string?
   (ds/opt :zip) string?
   (ds/opt :account_id) string?
   (ds/opt :account_lid) string?
   (ds/opt :address2) string?
   (ds/opt :affiliate_dock_description) string?
   (ds/opt :affiliate_dock_title) string?
   (ds/opt :affiliate_payment_method) string?
   (ds/opt :affiliate_paypal) string?
   (ds/opt :cc) string?
   (ds/opt :cc_auto) string?
   (ds/opt :cc_exp) string?
   (ds/opt :cc_type) string?
   (ds/opt :cc_whitelist) string?
   (ds/opt :ccs) account-info-data-ccs-spec
   (ds/opt :ccs_added) string?
   (ds/opt :company) string?
   (ds/opt :currency) string?
   (ds/opt :disable_reinstall) string?
   (ds/opt :disable_reset) string?
   (ds/opt :email) string?
   (ds/opt :email_abuse) string?
   (ds/opt :email_settings) account-info-data-email-settings-spec
   (ds/opt :extra) account-info-data-extra-spec
   (ds/opt :facebook_id) string?
   (ds/opt :facebook_url) string?
   (ds/opt :firstname) string?
   (ds/opt :fraudrecord) account-info-data-fraudrecord-spec
   (ds/opt :github_id) string?
   (ds/opt :github_url) string?
   (ds/opt :google_id) string?
   (ds/opt :google_url) string?
   (ds/opt :innertell_id) string?
   (ds/opt :lastname) string?
   (ds/opt :locale) string?
   (ds/opt :maxmind) account-info-max-mind-response-spec
   (ds/opt :maxmind_score) string?
   (ds/opt :mb_id) string?
   (ds/opt :modernbill_id) string?
   (ds/opt :picture) string?
   (ds/opt :referrer_coupon) string?
   (ds/opt :reseller_markup) string?
   (ds/opt :username) string?
   (ds/opt :ssh_key) string?
   (ds/opt :ssh_key_wrapped) string?
   (ds/opt :api_key) string?
   (ds/opt :api_key_wrapped) string?
   (ds/opt :2fa_google_key) string?
   (ds/opt :2fa_google_enabled) boolean?
   (ds/opt :2fa_google) int?
   (ds/opt :2fa_google_split) string?
   (ds/opt :2fa_google_qr) string?
   })

(def account-info-data-spec
  (ds/spec
    {:name ::account-info-data
     :spec account-info-data-data}))
